### DISCUSSION TITLE
pwm: 0 and invert_heater (NO valves)

### DISCUSSION THREAD

**Discussion by ZdenekM** [ScratMan/HASmartThermostat#238] [Q&A]
Hi,

I have normally open valves on my radiators, controlled by light dimmers so I'm using `invert_heater: true`, which used to work with on/off regulation. Now I tried to switch to `pwm: 0` but it seems that the `invert_heater: true` gets ignored in this case. Is that so? Is it a bug, or it is not supposed to work like that?

**Comment by RuslanKADR** [comment]
I have a similar problem, did you manage to find a solution? @ZdenekM

**Reply by ZdenekM** [reply]
Unfortunately, no, so I stuck to on/off control.

**Reply by ScratMan** [reply]
That's not possible for thé moment.
May you need to have 0 for opening valve and 100 to close it, you may use an input number like `input_number.thermostat_output` as heater entity receiving the output of the thermostat, and use an automation to take this value and compute the new value to be applied to the valve:

```yaml
{{ abs(100 - (states("input_number.thermostat_output") | float(0))) }}
```

If it work, I'll add it to the code.
